pkg/transfers/inbound/cleanup.go

```diff
@@ -54,3 +78,43 @@ func deleteFilesOnRemote(logger log.Logger, agent upload.Agent, localDir, suffix
 	}
 	return el
 }
+
+// deleteEmptyFiles deletes all empty files that are older than after (time.Duration)
+func deleteEmptyFiles(logger log.Logger, agent upload.Agent, localDir, suffix string, now time.Time, after time.Duration) error {
+	baseDir := filepath.Join(localDir, suffix)
+	infos, err := ioutil.ReadDir(baseDir)
+	if err != nil {
+		return fmt.Errorf("reading %s: %v", baseDir, err)
+	}
+
+	var el base.ErrorList
+	for i := range infos {
+		fileInfo := infos[i]
+		path := filepath.Join(suffix, filepath.Base(infos[i].Name()))
+		if shouldDeleteEmptyFile(fileInfo, now, after) {
+			err := agent.Delete(path)
+			if err != nil {
+				el.Add(err)
+			} else {
+				logger.Log(fmt.Sprintf("deleted zero byte file %s", path))
+			}
+		} else {
+			logger.Log(fmt.Sprintf("zero byte file not deleted %s", path))
+		}
+	}
+
+	if el.Empty() {
+		return nil
+	}
+	return el
```

Comment:
I think you could call `return el.Err()` since it'll return `nil` if it's empty.

Reply:
Good to know. I like the explicit-ness of this approach.

Reply:
I like this because it's less code. If we change it in this method, we should probably change it in every method to be consistent. What do ya think @vxio?

Reply:
I'd recommend doing that in a separate PR down the line.